widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionController.kt

```diff
@@ -37,20 +35,18 @@ internal class EngagementCompletionController(
 
     private fun handleEngagementState(state: State) {
         when (state) {
-            State.FinishedCallVisualizer, State.FinishedOmniCore -> {
-                releaseResourcesUseCase()
-                _state.onNext(EngagementCompletionState.QueuingOrEngagementEnded)
+            is State.EngagementEnded -> {
+                if (state.onEnd != Engagement.ActionOnEnd.RETAIN) releaseResourcesUseCase()
```

Comment:
It looks a bit confusing to me. I would expect to see that we keep resources for `RETAIN` event, based on it's name.

Reply:
Good point. I will investigate this a bit more

Reply:
I double checked the `releaseResourcesUseCase` being called on `ActionOnEnd.RETAIN` and yes, it is expected. We want to reset the Glia chat view and related controllers to letter setup them again into new correct states.

Reply:
I see. Thank you for checking. I think it might be useful to add this info as a comment to the code. IMHO.